packages/rum-core/src/common/context.js

```diff
@@ -0,0 +1,132 @@
+/**
+ * MIT License
+ *
+ * Copyright (c) 2017-present, Elasticsearch BV
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+import Url from './url'
+import { PAGE_LOAD } from './constants'
+import { getPageMetadata, getServerTimingInfo } from './utils'
+
+/**
+ * Both Navigation and Resource timing level 2 exposes these below information
+ *
+ * for CORS requests without Timing-Allow-Origin header, transferSize & encodedBodySize will be 0
+ */
+function getResponseContext(perfTimingEntry) {
+  const {
+    transferSize,
+    encodedBodySize,
+    decodedBodySize,
+    serverTiming
+  } = perfTimingEntry
+
+  const respContext = {
+    transfer_size: transferSize,
+    encoded_body_size: encodedBodySize,
+    decoded_body_size: decodedBodySize
+  }
+  const serverTimingStr = getServerTimingInfo(serverTiming)
+  if (serverTimingStr) {
+    respContext.headers = {
+      'server-timing': serverTimingStr
+    }
+  }
+  return respContext
+}
+
+function getResourceContext(data) {
```

Comment:
Is there any particular reason why we shouldn't or can't capture destination info for resources as well?

Reply:
They would be pointing to CDN or some other third part urls. Thought they are not useful for SIEM since it cannot be an attack vector. WDYT?

Reply:
> They would be pointing to CDN or some other third part urls. Thought they are not useful for SIEM since it cannot be an attack vector. WDYT?

SIEM-wise, I'm not sure. Seems like it could be interesting to know which third-party services users are being sent to, in case that changes unexpectedly.

We do want to capture the `destination.service.*`, so we can display CDNs and such in the service map.

Reply:
Cool, will incorporate the fields for those as well.

Reply:
I think they might even be relevant for SIEM, e.g loading JS file from an unknown location.